pkg/network/node/networkpolicy.go

```diff
@@ -610,7 +612,7 @@ func (np *networkPolicyPlugin) updateNetworkPolicy(npns *npNamespace, policy *ne
 	oldNPP, existed := npns.policies[policy.UID]
 	npns.policies[policy.UID] = npp
 
-	changed := !existed || !reflect.DeepEqual(oldNPP.flows, npp.flows) || !reflect.DeepEqual(oldNPP.selectedIPs, npp.selectedIPs)
+	changed := !existed || !reflect.DeepEqual(oldNPP.flows, npp.flows) || !reflect.DeepEqual(oldNPP.selectedIPs, npp.selectedIPs) || !reflect.DeepEqual(oldNPP.matchesAllIPs, npp.matchesAllIPs)
```

Comment:
right, though you don't need to use `reflect.DeepEqual` for the new check because it's not an array. You can just say `oldNPP.matchesAllIPs != npp.matchesAllIPs`

Reply:
done